### ISSUE TITLE
New Rule Request: Disallow access to properties by omitting props

### ISSUE THREAD

**Issue by yoyo837** [vuejs/eslint-plugin-vue#2689]
**Please describe what the rule should do:**

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<script setup>
const props = defineProps({
  list: Array,
});
</script>

<template>
  <div>{{ list }}</div> <!-- invalid -->
  <div>{{ props.list }}</div> <!-- valid -- >
</template>
```

**Additional context**

**Comment by FloEdelmann** [comment]
Duplicate of #2172
